parsl/execution_provider/aws/aws.py

```diff
@@ -0,0 +1,308 @@
+#!/usr/bin/env python
+
+import os
+import pprint
+import json
+import time
+
+from parsl.execution_provider.execution_provider_base import ExecutionProvider
+import boto3
+from botocore.exceptions import ClientError
+
+
+WORKER_USERDATA = '''#!/bin/bash
+export JAVA=/usr/local/bin/jdk1.7.0_51/bin
+export SWIFT=/usr/local/bin/swift-trunk/bin
+export PATH=$JAVA:$SWIFT:$PATH
+export WORKER_LOGGING_LEVEL=TRACE
```

Comment:
We do not need java and swift. It is reasonable to add the python3, pip3, and parsl installs.
I believe we will assume that the userdata will contain enough setup to ensure that python3 and parsl is runnable on the node.